Tribler/Main/vwxGUI/channel.py

```diff
@@ -979,11 +979,11 @@ def startDownloadFromUrl(self, url, *args, **kwargs):
 
     def startDownloadFromMagnet(self, url, *args, **kwargs):
         try:
-            return TorrentDef.retrieve_from_magnet(self.guiutility.utility.session, url, self.AddTDef, timeout=300)
+            callback = lambda meta_info: self.AddTDef(TorrentDef.load_from_dict(meta_info))
+            self.guiutility.utility.session.lm.ltmgr.get_metainfo(url, callback, timeout=300)
```

Comment:
return True after the call has been made, and revert the False if the except fires.